msos/StackWalker.cs

```diff
@@ -0,0 +1,372 @@
+            var paramz = GetParams(frame, ENTER_CRITICAL_SECTION_FUNCTION_PARAM_COUNT);
+            var criticalSectionPtr = Convert(paramz[0]);
+            frame.Handles = new List<UnifiedHandle>();
+            frame.Handles.Add(new UnifiedHandle(criticalSectionPtr, UnifiedHandleType.CriticalSection));
+        }
+
+        protected override void DealWithMultiple(UnifiedStackFrame frame)
+        {
+            var paramz = GetParams(frame, WAIT_FOR_MULTIPLE_OBJECTS_PARAM_COUNT);
+            if (paramz.Count > 0)
+            {
+                frame.Handles = new List<UnifiedHandle>();
+
+                var HandlesCunt = BitConverter.ToUInt32(paramz[0], 0);
+                var HandleAddress = BitConverter.ToUInt32(paramz[1], 0);
+
+                EnrichUnifiedStackFrame(frame, HandlesCunt, HandleAddress);
+            }
+        }
+
+        protected override void DealWithSingle(UnifiedStackFrame frame)
+        {
+            var paramz = GetParams(frame, WAIT_FOR_SINGLE_OBJECT_PARAM_COUNT);
+            if (paramz.Count > 0)
+            {
+                var handle = Convert(paramz[0]);
+
+                frame.Handles = new List<UnifiedHandle>();
+                frame.Handles.Add(new UnifiedHandle(handle));
+            }
+        }
+    }
+
+    /// <summary>
+    /// Stack walker abstract class, defining stack walker interface
+    /// </summary>
+    abstract class StackWalkerStrategy
+    {
+        #region Constants
+
+        public const string WAIT_FOR_SINGLE_OBJECTS_FUNCTION_NAME = "WaitForSingleObject";
+        public const string WAIT_FOR_MULTIPLE_OBJECTS_FUNCTION_NAME = "WaitForMultipleObjects";
+        public const string ENTER_CRITICAL_SECTION_FUNCTION_NAME = "EnterCriticalSection";
+
+        protected const int ENTER_CRITICAL_SECTION_FUNCTION_PARAM_COUNT = 1;
+        protected const int WAIT_FOR_SINGLE_OBJECT_PARAM_COUNT = 2;
+        protected const int WAIT_FOR_MULTIPLE_OBJECTS_PARAM_COUNT = 4;
+
+        private const int MAXIMUM_WAIT_OBJECTS = 0x00000102;
+
+        #endregion
+
+        protected readonly ClrRuntime _runtime;
+
+        public StackWalkerStrategy(ClrRuntime runtime)
+        {
+            _runtime = runtime;
+        }
+
+
+        private List<byte[]> ReadFromMemmory(ulong startAddress, uint count)
+        {
+            List<byte[]> result = new List<byte[]>();
+            int sum = 0;
+            //TODO: Check if dfor can be inserted into the REadMemmory result (seems to be..)
+            for (int i = 0; i < count; i++)
+            {
+                byte[] readedBytes = new byte[IntPtr.Size];
+                if (_runtime.ReadMemory(startAddress, readedBytes, IntPtr.Size, out sum))
+                {
+                    result.Add(readedBytes);
+                }
+                else
+                {
+                    throw new Exception($"Accessing Unreadable memorry at {startAddress}");
+                }
+                //Advancing the pointer by 4 (32-bit system)
```

Comment:
This is a dangerous comment. If you really meant to advance by 4, unconditionally, then just put 4 in the code. However, `IntPtr.Size` is going to be 8 on x64.